test/KeyRegistry/KeyRegistry.t.sol

```diff
@@ -49,140 +59,225 @@ contract KeyRegistryTest is KeyRegistryTestSuite {
     function testFuzzAdd(
         address to,
         address recovery,
-        uint32 scheme,
+        uint32 keyType,
         bytes calldata key,
-        bytes calldata metadata
+        uint8 metadataType,
+        bytes memory metadata
     ) public {
+        keyType = uint32(bound(keyType, 1, type(uint32).max));
+        metadataType = uint8(bound(metadataType, 1, type(uint8).max));
+
         uint256 fid = _registerFid(to, recovery);
+        _registerValidator(keyType, metadataType);
 
         vm.expectEmit();
-        emit Add(fid, scheme, key, key, metadata);
+        emit Add(fid, keyType, key, key, metadataType, metadata);
+        vm.prank(to);
+        keyRegistry.add(keyType, key, metadataType, metadata);
+
+        assertAdded(fid, key, keyType);
+    }
+
+    function testFuzzAddRevertsUnregisteredValidator(
+        address to,
+        address recovery,
+        uint32 keyType,
+        bytes calldata key,
+        uint8 metadataType,
+        bytes memory metadata
+    ) public {
+        keyType = uint32(bound(keyType, 1, type(uint32).max));
+        metadataType = uint8(bound(metadataType, 1, type(uint8).max));
+
+        /* We set a validator for keyType 1, type 1 during setup. Remove it.*/
```

Comment:
we didn't call _registerValidator here, are we still actually setting the validator?

Reply:
We are, but it's happening implicitly during `setUp`. This was an artifact of an intermediate refactor, we should use `_registerValidator` explicitly everywhere.